test/test_selectors.py

```diff
@@ -681,24 +689,93 @@ def test_above_fd_setsize(self):
+        selector = self.make_selector()
+        self.assertIsInstance(selector, selectors.SelectSelector)
+
+    @skipUnless(hasattr(errno, 'ENOSYS'), 'Platform does not have errno.ENOSYS.')
+    def test_select_module_defines_does_not_implement_poll(self):
+        # This test is to make sure that if a platform defines
+        # a selector as being available but does not actually
+        # implement it (kennethreitz/requests#3906) does not
+        # make DefaultSelector() fail.
+
+        # Reset the _DEFAULT_SELECTOR value as if using for the first time.
+        selectors._DEFAULT_SELECTOR = None
+
+        had_poll = hasattr(select, 'poll')
+
+        # Remove all selectors except `select.select`.
+        self.setup_select_module('select')
+
+        # Now we're going to patch in a bad `poll`.
+        class BadPoll(object):
+            def poll(self, timeout):
+                raise OSError(errno.ENOSYS)
+
+        select.poll = BadPoll
+        if not had_poll:
+            self.addCleanup(delattr, select, 'poll')
+
+        selector = self.make_selector()
+        self.assertIsInstance(selector, selectors.SelectSelector)
+
+    @skipUnless(hasattr(errno, 'ENOSYS'), 'Platform does not have errno.ENOSYS.')
+    def test_select_module_defines_does_not_implement_epoll(self):
+        # Same as above test except with `select.epoll`.
+
+        # Reset the _DEFAULT_SELECTOR value as if using for the first time.
+        selectors._DEFAULT_SELECTOR = None
+
+        had_epoll = hasattr(select, 'epoll')
+
+        # Remove all selectors except `select.select`.
+        self.setup_select_module('select')
+
+        # Now we're going to patch in a bad `epoll`.
+        def bad_epoll(*args, **kwargs):
+            raise OSError(errno.ENOSYS)
+
+        select.epoll = bad_epoll
+        if not had_epoll:
+            self.addCleanup(delattr, select, 'epoll')
+
+        selector = self.make_selector()
+        self.assertIsInstance(selector, selectors.SelectSelector)
+
+
 @skipUnless(hasattr(selectors, "SelectSelector"), "Platform doesn't have a SelectSelector")
 class SelectSelectorTestCase(BaseSelectorTestCase):
-    SELECTOR = getattr(selectors, "SelectSelector", None)
+    def setUp(self):
+        self.setup_select_module('select')
 
 
 @skipUnless(hasattr(selectors, "PollSelector"), "Platform doesn't have a PollSelector")
 class PollSelectorTestCase(BaseSelectorTestCase, ScalableSelectorMixin):
-    SELECTOR = getattr(selectors, "PollSelector", None)
+    def setUp(self):
+        self.setup_select_module('poll')
 
 
 @skipUnless(hasattr(selectors, "EpollSelector"), "Platform doesn't have an EpollSelector")
 class EpollSelectorTestCase(BaseSelectorTestCase, ScalableSelectorMixin):
-    SELECTOR = getattr(selectors, "EpollSelector", None)
+    def setUp(self):
+        self.setup_select_module('epoll')
 
 
 @skipUnless(hasattr(selectors, "KqueueSelector"), "Platform doesn't have a KqueueSelector")
```

Comment:
So we don't need to change it but we could? `selectors.py` checks to see if `select` has `kqueue` before defining `KqueueSelector`.